template/php7/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# start with the official Composer image and name it
+FROM composer:1.7 AS composer
+
+# continue with the official PHP image
+FROM php:7.2-alpine
+
+# copy the Composer PHAR from the Composer image into the PHP image
+COPY --from=composer /usr/bin/composer /usr/bin/composer
+
+# Composer requirements
+ARG COMPOSER_AUTH='{}'
+ENV COMPOSER_AUTH=${COMPOSER_AUTH}
+
+# Add watchdog and packages
+RUN apk add --no-cache git && \
+    echo "Pulling watchdog binary from Github." && \
+    curl -sSL https://github.com/openfaas/faas/releases/download/0.9.4/fwatchdog > /usr/bin/fwatchdog && \
+    chmod +x /usr/bin/fwatchdog
+
+# create non-root user
+RUN addgroup -S app && adduser -S -g app app && \
+    mkdir -p /home/app
+
+# Import function
+WORKDIR /home/app
+COPY index.php ./
+COPY ./function ./function
+WORKDIR /home/app/function
+
+# Install php extensions
+RUN [[ -f php-extension.sh ]] && sh ./php-extension.sh
+
+# Cleanup
+RUN apk del git && \
```

Comment:
The ordering has changed here so `git` gets deleted before `composer install`, does Composer need git in any scenarios, I see it execing out to a local `git` binary [here](https://github.com/composer/composer/blob/master/src/Composer/Util/Git.php)? If it's worth keeping in then I'd just shift the cleanup to after the `composer install`, the extra `USER` switches aren't a big deal.

Reply:
Good spot, yeah, there's a few cases where you would need composer to have Git (e.g. private repo's, etc..)